restaurants/src/app/components/restaurantdetails/restaurantdetails.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { Restaurant } from "../../models/restaurants.model";
+import { RestaurantsService } from "src/app/services/restaurants.service";
+import { RestaurantsComponent } from "../restaurants/restaurants.component";
+import { Router } from "@angular/router";
+import { ActivatedRoute } from "@angular/router";
+import { Location } from "@angular/common";
+import { Observable} from "rxjs";
+import { switchMap } from "rxjs";
+@Component({
+    selector: 'app-restaurantdetails',
+    templateUrl: './restaurantdetails.component.html',
+    styleUrls: ['./restaurantdetails.component.scss']
+  })
+  export class RestaurantDetailsComponent implements OnInit{
+    restaurant: Restaurant | undefined;
+
+    constructor(
+        private router: Router,
+        private route: ActivatedRoute,
+        private restaurantService: RestaurantsService,
+        private location: Location) {
+    }
+
+    ngOnInit(): void {
+        this.getRestaurant();
+    }
+
+    getRestaurant(): void {
+        const id = String(this.route.snapshot.paramMap.get('id'));
+        this.restaurantService.getRestaurant(id)
+            .subscribe(Restaurant => this.restaurant = Restaurant);
+    }
+
+    goBack(): void {
+        this.location.back();
```

Comment:
What if someone shared with you the url of a specific restaurant's detail page, and they entered that link into their browser? ie if a user was on Youtube, got the link http://localhost:4200/restaurant/3 from a friend and pasted it into the same browser tab, and then went to that page, then clicked the go back button? If you do `this.location.back()`, does it bring them back to Youtube, or the /restaurants page? Which page do you want the go back button to bring them to? I'm guessing you want it to be the the restaurants page, and not whatever random page they were on before - in which case, you probably want to navigate to that page explicitly instead of using `.back()`

Reply:
done?